Sources/MapboxMaps/Foundation/Camera/FlyToCameraAnimator.swift

```diff
@@ -0,0 +1,117 @@
+import UIKit
+
+public class FlyToCameraAnimator: NSObject, CameraAnimator, CameraAnimatorInterface {
+
+    internal private(set) weak var delegate: CameraAnimatorDelegate?
+
+    public private(set) var owner: AnimationOwner
+
+    internal private(set) var flyToInterpolator: FlyToInterpolator?
+
+    internal private(set) var animationDuration: TimeInterval?
+
+    public private(set) var state: UIViewAnimatingState = .inactive
+
+    internal private(set) var startTime: Date?
+
+    internal private(set) var endTime: Date?
+
+    internal private(set) var finalCameraOptions: CameraOptions?
+
+    internal private(set) var animationCompletion: AnimationCompletion?
+
+    internal init(delegate: CameraAnimatorDelegate,
+                  owner: AnimationOwner = .custom(id: "fly-to")) {
+
+        self.delegate = delegate
+        self.owner = owner
+    }
+
+    deinit {
+        scheduleCompletionIfNecessary(position: .current)
+    }
+
+    internal func makeFlyToInterpolator(from initalCamera: CameraOptions, to finalCamera: CameraOptions, duration: TimeInterval? = nil, screenFullSize: CGSize) {
+
+        guard let flyTo = FlyToInterpolator(from: initalCamera,
+                                                        to: finalCamera,
+                                                        size: screenFullSize) else {
+            assertionFailure("FlyToInterpolator could not be created.")
+            return
+        }
+
+        var time = duration ?? -1.0
+
+        // If there was no duration specified, or a negative argument, use a default
+        if time < 0.0 {
+            time = flyTo.duration()
+        }
+
+        animationDuration = time
+        flyToInterpolator = flyTo
+        finalCameraOptions = finalCamera
+    }
+
+    public func stopAnimation() {
+        state = .stopped
+        flyToInterpolator = nil
+        scheduleCompletionIfNecessary(position: .current) // `current` represents an interrupted animation.
+    }
+
+    internal func startAnimation() {
```

Comment:
should we do anything to prevent this animation from being restarted?

Reply:
Eventually? Yes.. especially if we start fully exposing the `FlyToCameraAnimator`.

Reply:
We will figure this out once we start fully exposing the flyToCameraAnimator